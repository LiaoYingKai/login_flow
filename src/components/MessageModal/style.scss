@mixin container($height, $borderBottom) {
	height: $height;
	border-bottom: $borderBottom solid #d6d6d6;
	padding: 0 16px;
	font-size: 18px;
	display: flex;
	align-items: center;
}

.message-modal {
	position: absolute;
	z-index: 1;
	height: 100vh;
	width: 100vw;
	display: flex;
	justify-content: center;
	align-items: center;
	background-color: rgba(0,0,0, .75);
	overflow: hidden;
	&--hidden {
		display: none;
	}
	&__content {
		border-radius: 6px;
		box-sizing: border-box;
		background-color: #fff;
		width: 300px;
		height: 180px;
		display: flex;
		justify-content: space-between;
		flex-direction: column;
	}
	&__title {
		@include container(40px, 1px);
	}
	&__message {
		@include container(90px, 1px);
	}
	&__footer {
		@include container(50px, 0px);
		justify-content: flex-end;
		.button {
			width: auto;
			padding: 4px 16px;
			font-size: 16px;
		}
	}
}