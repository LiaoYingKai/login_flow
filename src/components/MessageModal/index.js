import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import Button from '../Button';
import './style.scss';

const propTypes = {
	className: PropTypes.string,
	isVisible: PropTypes.bool,
	title: PropTypes.string,
	message: PropTypes.string,
	onClickOk: PropTypes.func,
	okText: PropTypes.string,
};

const defaultProps = {
	isVisible: false,
	title: '通知',
	okText: '確認',
	onClickOk: () => {},
};

function MessageModal({
	className,
	isVisible,
	title,
	message,
	onClickOk,
	okText,
}) {
	return (
		<div className={cx(
			className,
			"message-modal",
			{ "message-modal--hidden": !isVisible, }
		)}>
			<div className="message-modal__content">
				<div className="message-modal__title">
					{title}
				</div>
				<div className="message-modal__message">
					{message}
				</div>
				<div className="message-modal__footer">
					<Button
						onClick={onClickOk}
					>
						{okText}
					</Button>
				</div>
			</div>
		</div>
	);
}

MessageModal.propTypes = propTypes;
MessageModal.defaultProps = defaultProps;

export default MessageModal;
