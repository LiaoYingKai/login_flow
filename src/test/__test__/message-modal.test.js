/* eslint-disable no-undef */
import React from 'react';
import MessageModal from '../../components/MessageModal';
import { shallow, mount, } from "enzyme";
import toJson from 'enzyme-to-json';

describe("MessageModal", () => {
	it("should handle default props", () => {
		const {
			isVisible,
			title,
			okText,
			onClickOk,
		} = MessageModal.defaultProps;

		expect(isVisible).toBe(false);
		expect(title).toBe('通知');
		expect(okText).toBe('確認');
		expect(onClickOk).toBeDefined();
		expect(onClickOk).toBeInstanceOf(Function);
	});
	
	it('should select by message-modal class', () => {
		const wrapper = shallow(<MessageModal/>);

		expect(wrapper.hasClass('message-modal')).toEqual(true);
	});

	it('should select by custom class', () => {
		const className = 'mock-class';
		const wrapper = shallow(<MessageModal className={className}/>);

		expect(wrapper.hasClass('mock-class')).toEqual(true);
	});

	it("should render corrent", () => {
		const className = 'mock-class';
		const onClickOk = () => {};
		const isVisible = false;
		const title = 'mock-title';
		const message = 'mock-message';
		const okText = 'mock-ok-text';

		const wrapper = shallow(<MessageModal
			className={className}
			onClickOk={onClickOk}
			isVisible={isVisible}
			title={title}
			message={message}
			okText={okText}
		/>);

		expect(toJson(wrapper)).toMatchSnapshot();
	});

	it("should mount in a full DOM", () => {
		const className = 'mock-class';
		const onClickOk = () => {};
		const isVisible = false;
		const title = 'mock-title';
		const message = 'mock-message';
		const okText = 'mock-ok-text';

		const wrapper = mount(<MessageModal
			className={className}
			onClickOk={onClickOk}
			isVisible={isVisible}
			title={title}
			message={message}
			okText={okText}
		/>);

		expect(wrapper.props().className).toEqual(className);
		expect(wrapper.props().onClickOk).toEqual(onClickOk);
		expect(wrapper.props().isVisible).toEqual(isVisible);
		expect(wrapper.props().title).toEqual(title);
		expect(wrapper.props().message).toEqual(message);
		expect(wrapper.props().okText).toEqual(okText);
	});
});

