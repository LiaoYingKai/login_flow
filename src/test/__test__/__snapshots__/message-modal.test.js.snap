// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageModal should render corrent 1`] = `
<div
  className="mock-class message-modal message-modal--hidden"
>
  <div
    className="message-modal__content"
  >
    <div
      className="message-modal__title"
    >
      mock-title
    </div>
    <div
      className="message-modal__message"
    >
      mock-message
    </div>
    <div
      className="message-modal__footer"
    >
      <Button
        onClick={[Function]}
        type="button"
      >
        mock-ok-text
      </Button>
    </div>
  </div>
</div>
`;
